caffe2/python/trt/transform.py

```diff
@@ -0,0 +1,105 @@
+## @package onnx
+#Module caffe2.python.trt.transform
+
+"""
+TensorRT related transformation
+Note that ONNX-TRT enforce an NCHW input!
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from caffe2.proto import caffe2_pb2
+from caffe2.python.onnx.helper import c2_native_run_net, c2_native_run_op
+import caffe2.python.onnx.frontend as c2_front
+import caffe2.python._import_c_extension as C
+import numpy as np
+
+def _dim_values_to_list(dim_values):
+    return [x.dim_value for x in dim_values]
+
+
+def _get_output_shapes(output_value_infos):
+    names = [x.name for x in output_value_infos]
+    shapes = [_dim_values_to_list(x.type.tensor_type.shape.dim) for x in output_value_infos]
+    return dict(zip(names, shapes))
+
+
+def convert_onnx_model_to_trt_op(onnx_model,
+        max_batch_size=50,
+        max_workspace_size=2*1024*1024,
+        verbosity=1,
+        debug_builder=False):
+    """
+    Convert the whole ONNX model to a TensorRT C2 op
+    """
+    trt_str = C.onnx_to_trt_op(onnx_model.SerializeToString(),
+                               _get_output_shapes(onnx_model.graph.output),
+                               max_batch_size,
+                               max_workspace_size,
+                               verbosity,
+                               debug_builder)
+    op = caffe2_pb2.OperatorDef()
+    op.ParseFromString(trt_str)
+    return op
+
+def _infer_shapes(init_net, pred_net, inputs):
```

Comment:
you can also try to invoke C2's shape inference (which is spotty though). See e.g. https://github.com/onnx/onnx-coreml/blob/4aa0b5a819418acc2495be629085f97f32e65208/onnx_coreml/_shapeinference.py#L31

Reply:
Yeah, this is just a backup just in case C2's shape inference is spotty. I already used that.